problem_two.rb

```diff
@@ -0,0 +1,7 @@
+# Given an integer array of size N, identify the largest 2 numbers in this array. Do not sort the array.
+
+class ProblemTwo
+  def max_two(array)
+    array.select{ |x| x.is_a? Integer }.max(2)
```

Comment:
Furthermore if you had letters in the array for example `['a', 'b', 'c']` will it work without your `is_a?` check?

Reply:
Using `max` is risky, one criteria was that you should not sort the array, are you sure max does not actually sort the array in the background?

Reply:
> Furthermore if you had letters in the array for example `['a', 'b', 'c']` will it work without your `is_a?` check?

It would work if we would remove the `select` check, which makes sure only numbers are taken into account, as @drqCode pointed out in https://github.com/Agilefreaks/Test/pull/41, provided the input consists of letters only. `max` returns the letters based on their ASCII values, meaning `max(2) => ['c', 'b']`.

Without the check it would not, however, work on a mixed array such as `['a', 'b', 'c', 100, 101, 99]`.

> Using `max` is risky, one criteria was that you should not sort the array, are you sure max does not actually sort the array in the background?

Based on the [source code ](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-max)of the method and further research, I found no hint that `max` would sort the array, making it safe for use & an elegant solution.